docs/reference/components/sources/datadog_logs.cue

```diff
@@ -0,0 +1,115 @@
+package metadata
+
+components: sources: datadog_logs: {
+	_port: 8080
+
+	title: "Heroku Logplex"
+
+	description: """
+		Receives logs from a local Datadog Agent.
+		"""
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["aggregator"]
+		development:   "beta"
+		egress_method: "batch"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.datadog_logs
+
+				interface: socket: {
+					direction: "incoming"
+					port:      _port
+					protocols: ["http"]
+					ssl: "optional"
+				}
+			}
+
+			tls: {
+				enabled:                true
+				can_enable:             true
+				can_verify_certificate: true
+				enabled_default:        false
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		address:          sources.http.configuration.address
```

Comment:
I could see us adding an `api_key` configuration option as well to reject message without a matching API key; this would be similar to our `splunk_hec`'s `token` option. Not blocking though, this seems fine without it to start.

Reply:
I just check the `splunk_hec` / `token` option, the same kind of logic would definitely be a good fit for this new source. I will create an entry in the backlog for that.